pandas/tests/io/parser/test_network.py

```diff
@@ -31,24 +31,18 @@ def setup(self):
                          'pandas/io/tests/parser/data/salaries.csv')
 
     @tm.network
-    def test_compressed_urls(self):
-        # Test reading compressed tables from URL.
-        msg = ('Test reading {}-compressed tables from URL: '
-               'compression="{}", engine="{}"')
-
-        for compression, extension in self.compression_to_extension.items():
-            url = self.base_url + extension
-            # args is a (compression, engine) tuple
-            for args in product([compression, 'infer'], ['python', 'c']):
-                # test_fxn is a workaround for more descriptive nose reporting.
-                # See http://stackoverflow.com/a/37393684/4651668.
-                test_fxn = functools.partial(self.check_table)
-                test_fxn.description = msg.format(compression, *args)
-                yield (test_fxn, url) + args
-
-    def check_table(self, url, compression, engine):
-        if url.endswith('.xz'):
+    @pytest.mark.parametrize('compression, extension',
+                             compression_to_extension.items())
+    @pytest.mark.parametrize('mode', ['explicit', 'infer'])
+    @pytest.mark.parametrize('engine', ['python', 'c'])
+    def test_compressed_urls(self, compression, extension, mode, engine):
+        if extension == '.xz':
```

Comment:
oh you changed this too. fine with your changes as well (of course the tm.network issue still there though).